while true do
	shell.run("main")
	print("Main terminated, recovering")
end